src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -658,136 +707,157 @@ void func_80A75A38(EnIk* this, PlayState* play) {
+                prevInvincibilityTimer = player->invincibilityTimer;
+
+                if (player->invincibilityTimer <= 0) {
+                    if (player->invincibilityTimer < -39) {
+                        player->invincibilityTimer = 0;
+                    } else {
+                        player->invincibilityTimer = 0;
+                        play->damagePlayer(play, -64);
+                        this->unk_2FE = 0;
+                    }
                 }
+
+                func_8002F71C(play, &this->actor, 8.0f, this->actor.yawTowardsPlayer, 8.0f);
+                player->invincibilityTimer = prevInvincibilityTimer;
             }
-            func_8002F71C(play, &this->actor, 8.0f, this->actor.yawTowardsPlayer, 8.0f);
-            player->invincibilityTimer = prevInvincibilityTimer;
         }
-    }
-    Actor_MoveForward(&this->actor);
-    Actor_UpdateBgCheckInfo(play, &this->actor, 75.0f, 30.0f, 30.0f,
-                            UPDBGCHECKINFO_FLAG_0 | UPDBGCHECKINFO_FLAG_2 | UPDBGCHECKINFO_FLAG_3 |
-                                UPDBGCHECKINFO_FLAG_4);
-    this->actor.focus.pos = this->actor.world.pos;
-    this->actor.focus.pos.y += 45.0f;
-    Collider_UpdateCylinder(&this->actor, &this->bodyCollider);
-    CollisionCheck_SetOC(play, &play->colChkCtx, &this->bodyCollider.base);
-    if ((this->actor.colChkInfo.health > 0) && (this->actor.colorFilterTimer == 0) && (this->unk_2F8 >= 2)) {
-        CollisionCheck_SetAC(play, &play->colChkCtx, &this->bodyCollider.base);
-    }
-    if (this->unk_2FE > 0) {
-        CollisionCheck_SetAT(play, &play->colChkCtx, &this->axeCollider.base);
-    }
-    if (this->unk_2F8 == 9) {
-        CollisionCheck_SetAC(play, &play->colChkCtx, &this->shieldCollider.base);
+
+        Actor_MoveForward(&this->actor);
+        Actor_UpdateBgCheckInfo(play, &this->actor, 75.0f, 30.0f, 30.0f,
+                                UPDBGCHECKINFO_FLAG_0 | UPDBGCHECKINFO_FLAG_2 | UPDBGCHECKINFO_FLAG_3 |
+                                    UPDBGCHECKINFO_FLAG_4);
+
+        this->actor.focus.pos = this->actor.world.pos;
+        this->actor.focus.pos.y += 45.0f;
+
+        Collider_UpdateCylinder(&this->actor, &this->bodyCollider);
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->bodyCollider.base);
+
+        if ((this->actor.colChkInfo.health > 0) && (this->actor.colorFilterTimer == 0) && (this->unk_2F8 >= 2)) {
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->bodyCollider.base);
+        }
+
+        if (this->unk_2FE > 0) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->axeCollider.base);
+        }
+
+        if (this->unk_2F8 == 9) {
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->shieldCollider.base);
+        }
     }
 }
 
-Gfx* func_80A761B0(GraphicsContext* gfxCtx, u8 primR, u8 primG, u8 primB, u8 envR, u8 envG, u8 envB) {
+Gfx* EnIk_SetColors(GraphicsContext* gfxCtx, u8 primR, u8 primG, u8 primB, u8 envR, u8 envG, u8 envB) {
```

Comment:
I think `EnIk_SetPrimEnvColors` would be nice as a reminder of the arguments order, and it helps showing it's a "make a dlist" function ?

Reply:
imo you can get that info from the args pretty easily (either as your typing them or hovering over it). It just seemed more elegant and less of a mouthful

but yeah I dont care that much